plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -244,6 +224,15 @@ export class PersonState {
     }
 
     private applyEventPropertyUpdates(personProperties: Properties): Properties {
+        // Ideally we'd keep track of event timestamps, for when properties were updated
+        // and only update the values if a newer timestamped event set them.
+        // However to do that we would need to keep track of previous set timestamps,
+        // which means that even if the property value didn't change
+        // we would need to trigger an update to update the timestamps.
+        // This can kill Postgres if someone sends us lots of events with unchanged properties
+        // like our posthog-js library does.
+        // So instead we just process properties updates based on ingestion time,
+        // i.e. always update if value has changed.
```

Comment:
Nice comment. I do think we can build a solution for this further down the line.

Reply:
Learning from the best ... Harry comments ;)